File: app/src/java/java/com/longtoast/bilbil_api/dto/SearchHistoryDTO.java
// src/main/java/com/longtoast/bilbil_api/dto/SearchHistoryDTO.java
package com.longtoast.bilbil_api.dto;

import com.longtoast.bilbil_api.domain.UserSearchHistory;
import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Builder
public class SearchHistoryDTO {

    private String keyword;
    private LocalDateTime searchedAt;

    public static SearchHistoryDTO from(UserSearchHistory history) {
        return SearchHistoryDTO.builder()
                .keyword(history.getKeyword())
                .searchedAt(history.getSearchedAt())
                .build();
    }
}

File: app/src/main/java/com/longtoast/bilbil/dto/SearchHistoryDTO.kt
// com/longtoast/bilbil/dto/SearchHistoryDTO.kt
package com.longtoast.bilbil.dto

data class SearchHistoryDTO(
    val keyword: String,
    val searchedAt: String? // LocalDateTime 문자열로 들어올 거라 String 으로 받아도 충분
)
